qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/products/ProductsResourcesService.groovy

```diff
@@ -0,0 +1,53 @@
+package life.qbic.portal.qoffer2.products
+
+
+import life.qbic.datamodel.dtos.business.services.Product
+import life.qbic.portal.qoffer2.events.EventEmitter
+import life.qbic.portal.qoffer2.services.ResourcesService
+
+/**
+ * Product service that holds resources about available products
+ *
+ * This service holds a list of available products and exposes an
+ * event emitter, that can be used to subscribe to any update
+ * event of the underlying resource data.
+ *
+ * @since: 1.0.0
+ */
+class ProductsResourcesService implements ResourcesService {
+
+    private final ProductsDbConnector dbConnector
+    private final List<Product> products
+    /**
+     * EventEmitter for products. Fires every time the resources are reloaded
+     * @see #reloadResources
+     */
+    final EventEmitter<List<Product>> productEventEmitter
+
+
+    /**
+     *
+     * @param dbConnector
+     */
+    ProductsResourcesService(ProductsDbConnector dbConnector) {
+        this.dbConnector = dbConnector
+        this.products = new LinkedList<>()
+        this.productEventEmitter = new EventEmitter<>()
+        reloadResources()
+    }
+
+    @Override
+    void reloadResources() {
```

Comment:
Maybe inherit docs?

Reply:
Seems this is done automatically since java 9
https://docs.oracle.com/en/java/javase/15/docs/specs/javadoc/doc-comment-spec.html
Explicit inheritance is not needed if nothing changed to the implemented method.

Reply:
👍